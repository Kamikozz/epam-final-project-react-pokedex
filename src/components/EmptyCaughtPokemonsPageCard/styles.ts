import { createStyles } from "@material-ui/core";

const styles = createStyles({
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center"
  },
  cardContent: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  },
  text: {
    fontWeight: 400
  },
  caughtPokemonImage: {
    margin: "12px 0 0 0",
    width: "90%",
    borderRadius: "4px"
  }
});

export default styles;
