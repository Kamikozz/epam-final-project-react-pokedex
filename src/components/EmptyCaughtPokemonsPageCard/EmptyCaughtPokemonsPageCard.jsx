import React from "react";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";

import caughtPokemonImage from "../../assets/pokemon-catching.gif";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center"
  },
  cardContent: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  },
  text: {
    fontWeight: "400"
  },
  caughtPokemonImage: {
    margin: "12px 0 0 0",
    width: "90%",
    borderRadius: "4px"
  }
};

class EmptyCaughtPokemonsPageCard extends React.Component {
  render() {
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        <Card>
          <CardContent className={classes.cardContent}>
            <Typography variant="h5" className={classes.text}>
              Need to catch more pokemons
            </Typography>
            <img
              className={classes.caughtPokemonImage}
              src={caughtPokemonImage}
              alt="pokeball catches wild pokemon"
            />
          </CardContent>
        </Card>
      </div>
    );
  }
}

EmptyCaughtPokemonsPageCard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(EmptyCaughtPokemonsPageCard);
