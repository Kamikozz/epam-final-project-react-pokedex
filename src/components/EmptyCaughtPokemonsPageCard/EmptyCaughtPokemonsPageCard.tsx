import React from "react";
import PropTypes from "prop-types";

import { WithStyles } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";

import caughtPokemonImage from "../../assets/pokemon-catching.gif";
import styles from "./styles";

interface Props extends WithStyles<typeof styles> {
  classes: {
    root: string;
    cardContent: string;
    text: string;
    caughtPokemonImage: string;
  };
};

class EmptyCaughtPokemonsPageCard extends React.Component<Props> {
  render() {
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        <Card>
          <CardContent className={classes.cardContent}>
            <Typography variant="h5" className={classes.text}>
              Need to catch more pokemons
            </Typography>
            <img
              className={classes.caughtPokemonImage}
              src={caughtPokemonImage}
              alt="pokeball catches wild pokemon"
            />
          </CardContent>
        </Card>
      </div>
    );
  }
}

(EmptyCaughtPokemonsPageCard as React.ComponentClass<Props>).propTypes = {
  classes: PropTypes.object.isRequired
} as any;

export default withStyles(styles)(EmptyCaughtPokemonsPageCard);
