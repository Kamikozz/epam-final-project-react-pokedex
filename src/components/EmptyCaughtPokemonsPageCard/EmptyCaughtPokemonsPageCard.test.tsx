import React from 'react';
import { render } from '@testing-library/react';

import EmptyCaughtPokemonsPageCard from './EmptyCaughtPokemonsPageCard';

it('renders without crashing', () => {
  render(<EmptyCaughtPokemonsPageCard />);
});

it('matches snapshot', () => {
  const { container } = render(<EmptyCaughtPokemonsPageCard />);
  expect(container.firstChild).toMatchSnapshot()
});

