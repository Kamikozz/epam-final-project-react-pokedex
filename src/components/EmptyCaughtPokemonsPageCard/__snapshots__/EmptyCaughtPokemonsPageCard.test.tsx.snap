// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="EmptyCaughtPokemonsPageCard-root-70"
>
  <div
    class="MuiPaper-root-75 MuiPaper-elevation1-78 MuiPaper-rounded-76 MuiCard-root-74"
  >
    <div
      class="MuiCardContent-root-102 EmptyCaughtPokemonsPageCard-cardContent-71"
    >
      <h5
        class="MuiTypography-root-103 MuiTypography-h5-119 EmptyCaughtPokemonsPageCard-text-72"
      >
        Need to catch more pokemons
      </h5>
      <img
        alt="pokeball catches wild pokemon"
        class="EmptyCaughtPokemonsPageCard-caughtPokemonImage-73"
        src="pokemon-catching.gif"
      />
    </div>
  </div>
</div>
`;
